misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkLease.kt

```diff
@@ -85,6 +86,19 @@ internal class ZkLease(
     }
   }
 
+  override fun addListener(listener: Lease.StateChangeListener) {
+    lock.withLock {
+      listeners.add(listener)
+      when (status) {
+        // We already own the lease and should tell the listener
+        Status.HELD -> listener.afterAcquire(this)
+        // We don't know if we own the lease, so check
+        Status.UNKNOWN -> checkHeld()
```

Comment:
This is necessary since the client is no longer expected to call `checkHeld()` after requesting a lease.